src/object/ambient_sound.cpp

```diff
@@ -252,22 +126,91 @@ AmbientSound::collision(GameObject& other, const CollisionHit& hit_)
 void
 AmbientSound::draw(DrawingContext& context)
 {
-  if (Editor::is_active()) {
+  if (Editor::is_active())
     context.color().draw_filled_rect(m_col.m_bbox, Color(0.0f, 0.0f, 1.0f, 0.6f),
                                      0.0f, LAYER_OBJECTS);
-  }
 }
 
 void
 AmbientSound::stop_looping_sounds()
 {
-  stop_playing();
+  if (m_sound_source)
+    m_sound_source->stop(false);
 }
 
 void
 AmbientSound::play_looping_sounds()
 {
-  start_playing();
+  if (Editor::is_active())
+    return;
+
+  m_sound_source->play();
+}
+
+void
+AmbientSound::update(float dt_sec)
+{
+  const Player* const nearest_player = Sector::get().get_nearest_player(get_bbox().get_middle());
+  if (!nearest_player)
+    return;
+  const Rectf& player_bbox = nearest_player->get_bbox();
+  const Vector player_center = player_bbox.get_middle();
+
+  if (collision::intersects(player_bbox, get_bbox()))
```

Comment:
`player_bbox.contains(get_bbox())` might be better, since this is an old collision method, which is a duplicate of `Rectf::contains()` that, in my opinion, should be removed.

Reply:
Yeah, though according to FIXME in src/math/rectf.hpp it should be renamed to `overlaps` because name is misleading.

Reply:
That would be good to do while on it as well.